Form dialog (#2964)
microsoft/BotBuilder-Samples: experimental/generation/composer-form-dialog.md

# Form dialogs in Bot Framework Composer

The form dialogs feature in Composer uses [dialog generation][generation] to automatically create a dialog from a form. The form dialog collects information for fields of the form while handling ambiguity, corrections, help, navigation and multiple fields at once. With form dialogs, Composer uses your form and creates the adaptive dialog, the bot responses, and the language model for you.

**Note:** Form dialogs is a preview feature of Composer and should *not* be used in a production environment. To try an early version of form dialogs, follow the steps below.

## Enabling the form dialogs preview

To enable the form dialogs preview in Composer, follow these steps:

> 1. Install the [Bot Framework Composer][link to Composer downloads].

Author: - Added a link to how to get composer running...
- Changed the reference to Bot Framework Composer (removed 'the' to make it more consistent with other Composer docs).

```suggestion
1. Install [Bot Framework Composer](https://docs.microsoft.com/en-us/composer/install-composer#build-composer-from-source).
```

Reviewer: Users can either build Composer from the source or download the Electron app version. Change link to the article: [https://docs.microsoft.com/en-us/composer/install-composer](https://docs.microsoft.com/en-us/composer/install-composer) (no anchor link)?

> 2. In Composer, go to your **Settings** by clicking on the gear icon at the bottom of the menu.

Author: A few things:

1. In the Composer docs do not use 'gear icon', so removing to be consistent.
1. Adding a step to explicitly explain how to run composer.
1. Combined steps 2 & 3. This is consistent with many other steps in Composer docs.
1. Specifying `navigation pane`, which is used consistently in Composer docs.
1. Changing numbering to use the standard of each number in the list being `1.`

```suggestion
1. Once installed, run Composer in your browser using the address http://localhost:3000.
1. Select **Settings** from the Composer menu, then **Application settings** in the navigation pane.
```

> 3. On Settings page, select **App settings** in the navigation.

Author: Combined this step with previous step.

```suggestion
```

> 4. In the **Preview features** section, select the **Form dialogs** checkbox. You should see a new icon appear in the menu. This is the **Forms** section.

Author: Note: the checkbox is labeled **Show Form Dialog**.

```suggestion
1. In the **Preview features** section, select the **Show Form Dialog** checkbox. You can now select the new menu item titled **Forms (preview)** from the Composer menu.
```

> Once enabled, the form dialog preview will remain enabled until de-select this checkbox.

Author: ```suggestion
Once the **Show Form Dialog** checkbox is selected, the form dialog preview will remain enabled until it is de-selected.
```

    1. In the menu, select the **Forms** icon. You will see this page:    
    2. Under **Schemas**, select the plus button to add a new schema. 
    3. In the **"Create a form dialog"** window, provide a name. This will be the name of your schema as well as the dialog generated from it.
    4. Under **Schemas**, select the schema you just created. The schema editor should appear on the right. 
    5. Select **Add property**.
    6. A new property card will appear. Provide a **name** and a **type**.
    7. **Drag and drop cards to change priority or to make a property optional.** By default, properties are set as `required` and prioritized by they order in the form. ***Required properties*** are properties that your bot will ask the user to provide. The ***priority*** refers to the order in which the bot will ask for the required properties. ***Optional properties*** are properties the bot accepts if given but does not ask for.

3. **Generate your dialog.** When your schema is complete, press the "Generate dialog" button. 

> 4. **Inspect your new form dialog**. You can navigate to your dialog by selecting the overflow menu (...) next to the name of the form. You can also use select the **Design** iconin the main menu to see your list of dialogs to view the form dialog. The form dialog will appear in this list. The triggers for the dialog will be grouped by property along with a group for form-wide triggers.

Reviewer 2: >iconin [](start = 175, length = 6)

icon in

5. **Connect your form dialog to the root dialog.**
You connect your form dialog to your main dialog as you would any child dialog. For example, if you'd like to invoke your form dialog from the root, you can add a new trigger:
    1. From your root dialog, select the **Add** dropdown and select the **Add a new trigger**.
    2. In the **"Create a trigger"** window, select the trigger type **"Intent recognized"**.
   
       If your root dialog is using the default recognizer, add example trigger phrases. Add a condition `=turn.recognized.score > 0.8` to prevent your form from restarting.
    3. In your new trigger, add the action node **Manage properties > Begin a new dialog** and select your form dialog.

> Alternatively, you can add a **Begin a new dialog** node as an action in an existing trigger.

Reviewer 2: I would simplify this and indicate you can just call begin dialog to invoke the form or alternatively create a trigger.  For creating a trigger might be better to describe how you could use the whole form model.
